src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,35 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        dublicateCheck(user);
+        loginLegthCheck(user.getLogin());
+        passwordLegthCheck(user.getPassword());
+        ageCheck(user.getAge());
+        storageDao.add(user);
+        return user;
+    }
+
+    public void dublicateCheck(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("User with login: " + user.getLogin() + " already exist.");
+        }
+    }
+
+    public void loginLegthCheck(String login) {
+        if (login.length() < 6) {
```

Comment:
[checklist](https://github.com/mate-academy/jv-registration-validation/blob/master/checklist.md)
Move all hardcoded values to [constant fields](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names) and give them informative names.

Reply:
not fixed